sbnci/Modules/TPCSim/TPCSimValidation_module.cc

```diff
@@ -0,0 +1,246 @@
+    vector<short> fADC; ///<ADC vector element number i; no decompression is applied.
+    vector<raw::ChannelID_t> fChannel; ///<DAQ channel this raw data was read from.
+    vector<ULong64_t> fSamples; ///<Number of samples in the uncompressed ADC data.
+    vector<float> fPedestal; 
+    vector<float> fSigma;
+//    vector<raw::Compress_t> fCompression; ///<Compression algorithm used to store the ADC counts.
+
+}; //end def class TPCSimValidation
+
+//////////////////////////////////////////////////////////////////////////////////////
+ana::TPCSimValidation::TPCSimValidation(const fhicl::ParameterSet& pset) : 
+  EDAnalyzer(pset),
+  numevents(0)
+{
+
+  fGenieGenModuleLabel = pset.get<string>("GenieGenModuleLabel","generator");
+  fLArGeantModuleLabel = pset.get<string>("LArGeantModuleLabel","largeant");
+  fRawDigitLabel = pset.get<string>("RawDigitLabel","daq");
+  fVerbose             = pset.get<bool>("Verbose",false); 
+
+} //end constructor
+
+///////////////////////////////////////////////////////////////////////////////
+void ana::TPCSimValidation::beginJob() {
+  
+  fTree = tfs->make<TTree>("tpcsimTree", "Tree with TPC validation information");
+  //gInterpreter->GenerateDictionary("vector<vector<float> > ","vector");
+
+  fTree->Branch("Run",          &fRun,          "Run/I");
+  fTree->Branch("SubRun",       &fSubRun,       "SubRun/I");
+  fTree->Branch("Event",        &fEvent,        "Event/I");
+  fTree->Branch("NRawDigits",       &fNRawDigits,       "NRawDigits/I");
+//  fTree->Branch("ADCs",&fADCs);
+  fTree->Branch("NADC",&fNADC);
+  fTree->Branch("ADC", &fADC);
+  fTree->Branch("Channel", &fChannel);
+  fTree->Branch("Samples", &fSamples);
+  fTree->Branch("Pedestal", &fPedestal);
+  fTree->Branch("Sigma", &fSigma);
+//  fTree->Branch("Compression", &fCompression);
+
+}// end beginJob
+
+////////////////////////////////////////////////////////////////////
+void ana::TPCSimValidation::analyze(const art::Event& evt) {
+
+  fRun    = evt.run();
+  fSubRun = evt.subRun();
+  fEvent  = evt.event();
+
+  if(fVerbose) {
+      cout << "Analysing run " << fRun << ", subrun " << fSubRun << ", event: " << fEvent << endl;
+  }
+  numevents++;
+
+  //Getting  MC truth information
+  art::Handle< vector<simb::MCTruth> > mctruthListHandle;
+  vector<art::Ptr<simb::MCTruth> > mclist;
+  if(evt.getByLabel(fGenieGenModuleLabel,mctruthListHandle)){
+      art::fill_ptr_vector(mclist, mctruthListHandle);
+  }
+
+  //###############################################
+  //### Get the Truth information for the event ###
+  //###############################################
+
+  //List the particles in the event
+  const sim::ParticleList& particles = particleInventory->ParticleList();
+
+  //Loop over the particles
+  map<int,const simb::MCParticle*> trueParticles;
+  map<int,bool> mcparticlescontained;
+  map<int,float> trueParticleEnergy;
+
+  //auto const clockData = art::ServiceHandle<detinfo::DetectorClocksService const>()->DataFor(evt);
```

Comment:
As before

Reply:
probably not needed. if the experts need this, they can always add it back.